### ISSUE TITLE
Suggest a new way to avoid the bot that is attempting to login 

### ISSUE THREAD

**Issue by jcockroft64** [jcockroft64/sierrahiking#81]
We are constantly getting attacked by a bot that is attempting to guess our passwords.
I have a plugin that locks the account after too many failed logins.   Is there another way to protect the site from such attacks?

5 failed login attempts (1 lockout(s)) from IP: 172.31.4.35

Last user attempted: jason

IP was blocked for 20 minutes

**Comment by russellelliott** [comment]
The main issue with the site is that there is no means of distinguishing between a human and bot by entering credential information alone. Therefore, security features that only humans can interact with must be implemented to ensure users aren't locked out of the site. 

Here are some security features we can look into to address the current issues:

## 2 Factor Authentication
This relates to implementing social media plugins @82 in that when login in with your social media profile, an email is sent for the user to verify their identity. There's also a plugin that does this for the existing native Wordpress site profiles. 2 factor authentication codes can be sent over email for users to enter upon logging in. Since it requires a successful login, and for the true user to enter the code, it may interfere with the bot's effectiveness in our favor.

Website: https://www.wpbeginner.com/plugins/how-to-add-two-factor-authentication-for-wordpress/
Plugin Link: https://www.wpbeginner.com/plugins/how-to-add-two-factor-authentication-for-wordpress/
Youtube Tutorial: https://www.youtube.com/watch?v=vRlX_NNGeFo

## Finding and Blocking Bots
https://onlinemediamasters.com/block-bad-bots-wordpress/
This site outlines various means/plugins for blocking bots.

View Bot traffic with WordFence; useful for creating a list of spam IP addresses
WordFence is a plugin that can be used to collect spam IP addresses. From there, they can be blocked
https://wordpress.org/plugins/wordfence/
https://www.wordfence.com/help/tools/live-traffic/

Free version: https://www.wordfence.com/products/wordfence-free/

Block by IP address: Go to blocking settings in WordFence to specify IP addresses or host names to block. You can also block a range of IP addresses.
Block by rate limiting: Go to rate limiting settings and adjust them to your preferences. This can be used to throttle unusually high traffic, typically from bots.

## Permanently Ban a Bot
The Blackhole plugin generates a link which will cause a bot to be permanently banned from the site upon accessing it. 
Free version: https://wordpress.org/plugins/blackhole-bad-bots/
Pro version: https://plugin-planet.com/blackhole-pro/
Good bots follow these rules and humans are unable to see/access the link. Bad bots will click the link and be banned forever.
The pro version adds more customization for the email messages, redirects, status codes, etc.

**Comment by russellelliott** [comment]
I installed a 2FA plugin
https://wp2fa.io/free-plugin-download/

I Set up 2FA with setup wizard
- Code via email
- Enforced for all users
- No users/roles excluded

Here are some other important things part of the plugin:

Grace period: users must enable 2 factor authentication within the next 7 days. Once you are authenticated we can adjust this setting. This setting can be changed in the 2FA policies settings.

If a user doesn't enable 2FA within the grace period of 7 days, their account is locked and a site administrator would have to unlock their account. View more info here: https://wp2fa.io/support/kb/configure-2fa-policies-enforce/?utm_source=plugin&utm_medium=referral&utm_campaign=WP2FA&utm_content=settings+pages

Backup codes: A user can generate a list of backup codes upon logging in. Backup codes are a secondary method which you can use to log in to the website in case the primary 2FA method is unavailable. Therefore they can't be enabled and used as a primary method.

2 Factor authentication would be effective against the bot, as the bot wouldn't be able to login with the username and password without the authentication code sent to the user's email. User will be prompted to enter an authentication code every time they login

Here's more info: https://www.wpbeginner.com/plugins/how-to-add-two-factor-authentication-for-wordpress/

Mandatory 2FA: https://wp2fa.io/support/kb/configure-2fa-policies-enforce/?utm_source=plugin&utm_medium=referral&utm_campaign=WP2FA&utm_content=settings+pages

**Comment by russellelliott** [comment]
# More info about Wordfence
https://www.wordfence.com/products/pricing#compare-plans
- block by IP (premium)
- rate limiting; limit amount of requests, could be useful against bruteforce bot
- 2 Factor Authentication and RECAPTCHA

You need to get a Wordfence license sent to your email upon activating the plugin.

# Blocking by Host Name
There's also the act of blocking by hostname, which could prove useful. This can be done by installing the Defender Security plugin and modifying the `.htaccess` by listing the host names to block.
Forum post: https://wordpress.org/support/topic/option-to-block-host-name/
Plugin in question: https://wordpress.org/plugins/defender-security/

As the forum post explains, the bot could use one or more host name, but multiple IP addresses. Blocking the host name could prove more effective than blocking the individual IP addresses, as the amount of host names would be signifigantly less the amount of IP addresses.

Here are some other features the plugin provides
- block bot IP addresses
- 2 Factor Authentication and RECAPTCHA

**Comment by russellelliott** [comment]
Did some more research concerning the firewall/bot blocker.

**Comment by russellelliott** [comment]
After extensive research, I can conclude that MalCare is the best choice.
https://blogvault.net/wordpress-security-plugins/
TLDR: [MalCare](https://www.malcare.com/) beat every single WordPress security plugin on the list with its top-of-the-line scanner, intelligent firewall, and flawless one-click cleanups. MalCare’s accuracy and reliability make it the best available WordPress security plugin for your site.

https://www.malcare.com/blog/malcare-free-vs-premium/
Malcare's free plugin includes a firewall and thorough malware checks. You need the premium plan to remover malware and request security audits. Premium costs $99/year for premium features, including bot IP blocking.

I installed the plugin and connected it to the site. Seeing as how I was locked out for several hours yesterday, and was nearly locked out this morning, it's critical to have some sort of security system in place.

**Comment by jcockroft64** [comment]
We have had no attempted login lockouts in the last 24 hrs.   It looks like the Malcare works well.

**Comment by jcockroft64** [comment]
I'll keep this open for another 24 hrs just to monitor.

**Comment by jcockroft64** [comment]
I noted in the Firewall that the brute force attack (which the login is), was reported as blocked:
<img width="880" alt="Screen Shot 2022-12-28 at 8 05 33 PM" src="https://user-images.githubusercontent.com/53958343/209902107-82d9048c-d0a4-4148-9634-fa4235582a57.png">

**Comment by jcockroft64** [comment]
This is working perfectly!
